pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -199,6 +204,42 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 	}
 	req.instance = instance
 
+	res, err := r.doReconcile(req)
+
+	/*
+			From K8s API reference: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/
+		    ============================================================================================================
+			Replace: Replacing a resource object will update the resource by replacing the existing spec with the
+			provided one. For read-then-write operations this is safe because an optimistic lock failure will occur if
+			the resource was modified between the read and write.
+
+			**Note: The ResourceStatus will be ignored by the system and will not be updated. To update the status, one
+			must invoke the specific status update operation.**
+			============================================================================================================
+
+			So we need to update both the CR and the CR Status.
+	*/
+	if req.dirty {
+		updateErr := r.client.Update(req.ctx, req.instance)
+		if updateErr != nil {
+			req.logger.Info("failed to update the CR", updateErr)
+			err = updateErr
+		}
+	}
+
+	if req.statusDirty {
+		updateErr := r.client.Status().Update(req.ctx, req.instance)
```

Comment:
Maybe I'm wrong, 
but we read instance from k8s only once on line 201.
Then, if we already wrote on k8s on line 223, the object on k8s has now a resourceVersion which is higher then what we have on req.instance and so this write is going to be rejected.